File: src/main/java/com/github/nsc/de/compiler/parser/node/expression/PowNode.java
package com.github.nsc.de.compiler.parser.node.expression;

import com.github.nsc.de.compiler.parser.node.Node;

public class PowNode extends ExpressionNode {
    public PowNode(Node left, Node right) {
        super(left, right);
    }

    @Override
    public char getOperator() {
        return '^';
    }
}

File: parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/expression/PowNode.kt
package io.github.shakelang.shake.parser.node.expression

import io.github.shakelang.shake.parser.node.ValuedNode
import io.github.shakelang.parseutils.characters.position.PositionMap

class PowNode(map: PositionMap, left: ValuedNode, right: ValuedNode, operatorPosition: Int) :
    ExpressionNode(map, left, right, operatorPosition) {
    override val operator: Char
        get() = '^'

    override fun toJson(): Map<String, *> = mapOf("name" to "PowNode", "left" to left, "right" to right)
}